tasks/databases-docker.yml

```diff
@@ -0,0 +1,93 @@
+  register: uuid_ext_result
+  failed_when: uuid_ext_result.rc != 0 and ("already exists, skipping" not in uuid_ext_result.stderr)
+  changed_when: uuid_ext_result.rc == 0 and ("already exists, skipping" not in uuid_ext_result.stderr)
+  when: item.uuid_ossp is defined and item.uuid_ossp
+  environment:
+    PGPASSWORD: "{{postgresql_admin_pwd}}"
+
+- name: PostgreSQL | Add postgis to the databases with the requirement
+  become: yes
+  shell: "{{ postgresql_bin_directory}}/psql {{item.name}} --username {{postgresql_admin_user}} --host {{postgresql_login_host}} --port {{postgresql_port}} -c 'CREATE EXTENSION IF NOT EXISTS postgis;'&&psql {{item.name}} --username {{postgresql_admin_user}} --host {{postgresql_login_host}} --port {{postgresql_port}} -c 'CREATE EXTENSION IF NOT EXISTS postgis_topology;'"
+  with_items: "{{postgresql_databases}}"
+  when: item.gis is defined and item.gis
+  environment:
+    PGPASSWORD: "{{postgresql_admin_pwd}}"
+
+- name: PostgreSQL | add cube to the database with the requirement
+  become: yes
+  shell: "{{ postgresql_bin_directory}}/psql {{item.name}} --username {{ postgresql_admin_user }} --host {{postgresql_login_host}} --port {{postgresql_port}} -c 'create extension if not exists cube;'"
+  with_items: "{{postgresql_databases}}"
+  when: item.cube is defined and item.cube
+  environment:
+    PGPASSWORD: "{{postgresql_admin_pwd}}"
+
+- name: PostgreSQL | Add plpgsql to the database with the requirement
+  become: yes
+  shell: "{{ postgresql_bin_directory}}/psql {{item.name}} --username {{ postgresql_admin_user }} --host {{postgresql_login_host}} --port {{postgresql_port}} -c 'CREATE EXTENSION IF NOT EXISTS plpgsql;'"
+  with_items: "{{postgresql_databases}}"
+  when: item.plpgsql is defined and item.plpgsql
+  environment:
+    PGPASSWORD: "{{postgresql_admin_pwd}}"
+
+- name: PostgreSQL | add earthdistance to the database with the requirement
+  become: yes
+  shell: "{{ postgresql_bin_directory}}/psql {{item.name}} --username {{ postgresql_admin_user }} --host {{postgresql_login_host}} --port {{postgresql_port}} -c 'create extension if not exists earthdistance;'"
+  with_items: "{{postgresql_databases}}"
+  when: item.earthdistance is defined and item.earthdistance
+  environment:
+    PGPASSWORD: "{{postgresql_admin_pwd}}"
+
+- name: PostgreSQL | Add citext to the database with the requirement
+  become: yes
+  shell: "{{ postgresql_bin_directory}}/psql {{item.name}} --username {{postgresql_admin_user}} --host {{postgresql_login_host}} --port {{postgresql_port}} -c 'CREATE EXTENSION IF NOT EXISTS citext;'"
+  with_items: "{{postgresql_databases}}"
+  register: citext_ext_result
+  failed_when: citext_ext_result.rc != 0 and ("already exists, skipping" not in citext_ext_result.stderr)
+  changed_when: citext_ext_result.rc == 0 and ("already exists, skipping" not in citext_ext_result.stderr)
+  when: item.citext is defined and item.citext
+  environment:
+    PGPASSWORD: "{{postgresql_admin_pwd}}"
```

Comment:
I diff'ed the two files (docker and default), while I agree that there is a lot of duplication, to keep a single file we would need flags everywhere (docker/non-docker).

Imho, the best solution would be to either use a loop for the exentions or call an `include` to which the variables are passed in a loop. But I would do this change as an upstream PR, not here.